src/fparser/two/Fortran2008.py

```diff
@@ -96,6 +101,241 @@ class Program_Unit(Program_Unit_2003):  # R202
     subclass_names.append("Submodule")
 
 
+class Component_Attr_Spec(STRINGBase):  # R437
+    """
+:F08R:`437`
+    <component-attr-spec> = <access-spec>
+                            | ALLOCATABLE
+                            | CODIMENSION <lbracket> <coarray-spec> <rbracket>
+                            | CONTIGUOUS
+                            | DIMENSION ( <component-array-spec> )
+                            | POINTER
+
+    This rule adds CODIMENSION and CONTIGUOUS attributes to Fortran2003's R441.
+    """
+    subclass_names = ['Access_Spec', 'Dimension_Component_Attr_Spec',
+                      'Codimension_Attr_Spec']
+    use_names = []
+
+    @staticmethod
+    def match(string):
+        '''Implements the matching for component attribute specifications.
+
+        :param str string: the string to match as an attribute.
+        :return: None if there is no match, otherwise an object with the \
+            matched attribute.
+        :rtype: None, Component_Attr_Spec, Codimension_Attr_Spec, \
+            Access_Spec, Dimension_Component_Attr_Spec
+        '''
+        return STRINGBase.match(
+            ['ALLOCATABLE', 'CONTIGUOUS', 'POINTER'], string)
+
+
+class Codimension_Attr_Spec(WORDClsBase):  # R502.d
+    """
+    <codimension-attr-spec> = CODIMENSION <lbracket> <coarray-spec> <rbracket>
```

Comment:
Again, could you please modify the docstring format. I think this applies to subsequent changes as well but I stop mentioning it now :-)